qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,521 @@
+        i_i = 1 if i_i == 0 else 0  # int(not(i_i))
+        if edge_list[i_i][j_j] < j:
+            v[j_j] = 1
+
+    qubit_position_j = np.asarray(np.where(edge_list == j))
+    for edge_index in range(qubit_position_j.shape[1]):
+        i_i, j_j = qubit_position_j[:, edge_index]
+        i_i = 1 if i_i == 0 else 0  # int(not(i_i))
+        if edge_list[i_i][j_j] < i:
+            v[j_j] = 1
+
+    qubit_op = Pauli((v, w))
+    return SparsePauliOp(qubit_op)
+
+
+def edge_operator_bi(edge_list: np.ndarray, i: int) -> SparsePauliOp:
+    """Calculate the edge operator B_i.
+
+    The definitions used here are consistent with arXiv:quant-ph/0003137
+
+    Args:
+        edge_list (numpy.ndarray): a 2xE matrix, where E is total number of edge
+                                    and each pair denotes (from, to)
+        i (int): index for specifying the edge operator B.
+
+    Returns:
+        Pauli: qubit operator
+    """
+    qubit_position_matrix = np.asarray(np.where(edge_list == i))
+    qubit_position = qubit_position_matrix[1]
+    v = np.zeros(edge_list.shape[1])
+    w = np.copy(v)
+    v[qubit_position] = 1
+    qubit_op = Pauli((v, w))
+    return SparsePauliOp(qubit_op)
+
+
+def _add_sparse_pauli(qubit_op1: SparsePauliOp, qubit_op2: SparsePauliOp) -> SparsePauliOp:
+    """
+    Return `qubit_op1` and `qubit_op2`, except when either one is `None`.
+    In the latter case, return the one that is not `None`. In other words, assume
+    `None` signifies the additive identity.
+    """
+    if qubit_op1 is None:
+        return qubit_op2
+    elif qubit_op2 is None:
+        return qubit_op1
+    else:
+        return qubit_op1 + qubit_op2
+
+
+def _to_physicist_index_order(facs: List) -> Tuple[List, int]:
+    """
+    Reorder the factors `facs` to be two raising operators followed by two lowering operators and
+    return the new factors and the phase incurred by the reordering. Note that `facs` are not in
+    chemists' order, but rather sorted by index with least index first.
+
+    Args:
+      facs: a list of factors where each element is `(i, c)` where `i` is an integer index and
+        `c` is either `-` or `+`.
+
+    Returns:
+        facs_out: A copy of the reordered factors or the input list (not a copy) if the factors are
+          already in the desired order.
+        phase: Either `1` or `-1`.
+
+    Raises:
+        ValueError: if `facs` does not represent a two-body interaction.
+    """
+    ops = [fac[1] for fac in facs]
+    if ops == ["+", "+", "-", "-"]:
+        facs_out = facs
+        phase = 1
+    elif ops == ["+", "-", "+", "-"]:
+        facs_out = [facs[0], facs[2], facs[1], facs[3]]
+        phase = -1
+    elif ops == ["+", "-", "-", "+"]:
+        facs_out = [facs[0], facs[3], facs[1], facs[2]]
+        phase = 1
+    else:
+        raise ValueError("unexpected sequence of operators", facs)
+    return facs_out, phase
+
+
+class BravyiKitaevSFMapper(FermionicMapper):
```

Comment:
```suggestion
class BravyiKitaevSuperFastMapper(FermionicMapper):
```
We tend to stick to full name so I'd argue we do the same here.

Reply:
Done